internal/dataplane/parser/translators/httproute.go

```diff
@@ -350,14 +352,35 @@ func mustMarshalJSON[T any](val T) string {
 	return string(key)
 }
 
-// GeneratePluginsFromHTTPRouteFilters converts HTTPRouteFilter into Kong plugins.
+func ConvertFiltersToPlugins(route *kongstate.Route, filters []gatewayapi.HTTPRouteFilter, path string, tags []*string) {
+	plugins, pluginAnnotation := generatePluginsFromHTTPRouteFilters(filters, path, tags)
+	if route.Plugins == nil {
+		route.Plugins = make([]kong.Plugin, 0)
+	}
+	route.Plugins = append(route.Plugins, plugins...)
+	if len(pluginAnnotation) > 0 {
+		if route.Ingress.Annotations == nil {
+			route.Ingress.Annotations = make(map[string]string)
+		}
+		if _, ok := route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey]; !ok {
+			route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey] = pluginAnnotation
+		} else {
+			route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey] = fmt.Sprintf("%s,%s",
+				route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey],
+				pluginAnnotation)
+		}
+	}
+}
+
+// generatePluginsFromHTTPRouteFilters converts HTTPRouteFilter into Kong plugins.
 // path is the parameter to be used by the redirect plugin, to perform redirection.
-func GeneratePluginsFromHTTPRouteFilters(filters []gatewayapi.HTTPRouteFilter, path string, tags []*string) []kong.Plugin {
+func generatePluginsFromHTTPRouteFilters(filters []gatewayapi.HTTPRouteFilter, path string, tags []*string) ([]kong.Plugin, string) {
```

Comment:
Can we add a godoc about the return values? The added return `string` is not obvious what it might contain